### DISCUSSION TITLE
"Can Build" propaged up through assemblies?

### DISCUSSION THREAD

**Discussion by jaminy5** [inventree/InvenTree#8046] [Q&A] [answered]
Currently, it appears that Can Build for a part is derived entirely from stock.  However, it would be useful if it could also include the Can Build of sub-assemblies that could be assembled from stock.  

I have several products that are assembled from sub-assemblies also manufactured in-house.  I typically keep a stock of sub-assembles, then assemble the sell-able products on demand.  For example, A (sellable product) is made from (among other things) B which is made from C which is made from on-hand stock.  Even if I have all parts necessary to create A, if I don't currently have B in-stock, then I'm not able to tell how many A could be made.

Is there a way to have 'Can Build' (or similar metric) carry all the way through sub-assemblies?  I can't think of another way to determine what non in-house-produced stock needs to be ordered without having to keep a ready stock of all sub-assemblies.

**Comment by matmair** [comment]
That is a very expensive roundtrip, we will probably not add this to the existing APIs.

**Comment by SchrodingersGat** [comment]
@jaminy5 this sort of information could be generated "on demand"… either using a custom report or via a plugin. Either approach allows direct database access where you can do more complex queries

**Reply by jaminy5** [reply, marked as answer]
A report would be perfect.  I'll take a look a look at reports and the plugin API.  If I can iterate the tree of BOMs under the top level part of interest, I'm sure I can get what I'm looking for.  I just didn't want to start down that path if there was an easier, non-obvious solution.

@matmair and @SchrodingersGat, thanks for the replies.

**Reply by cbenhagen** [reply]
@jaminy5 have you made any progress on this yet?

**Reply by matmair** [reply]
I have locked this as it was already answered;
@cbenhagen you can discuss this on https://github.com/inventree/InvenTree/issues/6665 but please do not rehash old discussions, it takes away from the notification management time.

**Comment by cbenhagen** [comment]
Fore reference, this is also discussed [here](https://github.com/inventree/InvenTree/discussions/5517) and [here](https://github.com/inventree/InvenTree/discussions/4201) but I can't find an open issue for this specific request.

**Reply by cbenhagen** [reply]
One closed issue exists though: https://github.com/inventree/InvenTree/issues/6665
